galsim_extra/mixed_scene.py

```diff
@@ -58,8 +59,23 @@ def setup(self, config, base, xsize, ysize, ignore, logger):
         # object ends up being chosen.
         ignore = ignore + ['objects', 'magnify', 'shear', 'obj_type']
 
+        stamp_xsize, stamp_ysize, image_pos, world_pos = super(MixedSceneBuilder, self).setup(config,base,xsize,ysize,ignore,logger)
+        # option to shear the full scene.
+        if config.get('shear_full_scene', True):            
+            shear = galsim.Shear(g1=float(base["stamp"]["shear"]["g1"]), g2=float(base["stamp"]["shear"]["g2"]))
+            S = shear.getMatrix()
+            # Find the center (tangent point) of the scene in RA, DEC. 
+            scene_center = base['wcs'].center
+            u,v = scene_center.project_rad(world_pos.ra, world_pos.dec, projection='gnomonic') # tile center units in radians
+            # shearing the position. 
+            pos = np.vstack((u, v))
+            sheared_uv = np.dot(S, pos)
+            # convert sheared u,v back to sheared ra,dec
+            sheared_ra, sheared_dec = scene_center.deproject_rad(sheared_uv[0,:].astype(float), sheared_uv[1,:].astype(float), projection='gnomonic')
+            world_pos = galsim.CelestialCoord(sheared_ra[0]*galsim.radians, sheared_dec[0]*galsim.radians)
+
```

Comment:
Another thing I just noticed.  At the end of this block, you've changed world_pos, but not image_pos.  That's likely to cause problems, since they won't correspond to the same location on the image.  You should fix image_pos now: `image_pos = wcs.toImage(world_pos)`.